File: Entrada de Dados/Numerica/numero.go
package main

import ("fmt"; "strconv")

var NUMERO int
var NUMERO_AUX string

func main() {
  for {
    fmt.Print("Entre um numero: ")
    fmt.Scanln(&NUMERO_AUX)
    if ACERTO, ERRO := strconv.Atoi(NUMERO_AUX); ERRO == nil {
      NUMERO = ACERTO
      break
    } else {
      fmt.Println("Entrada invalida. Por favor, tente novamente.")
    }
  }
  fmt.Printf("Voce informou o numero: %d\n", NUMERO)
}

File: Entrada de Dados/Numerica/numero.dart
import 'dart:io';

void main() {
  int NUMERO;
  do {
    stdout.write('Entre um número: ');
    try {
      NUMERO = int.parse(stdin.readLineSync()!);
      break;
    } catch (FormatException) {
      print('Entrada inválida. Por favor, tente novamente.');
    }
  } while (true);
  print('Você informou o número: $NUMERO');
}
